vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -141,19 +141,29 @@ We cannot verify .NET is safe to download at this time. Please try again later.`
      */
     private async downloadInstaller(installerUrl : string) : Promise<string>
     {
-        const ourInstallerDownloadFolder = IGlobalInstaller.getDownloadedInstallFilesFolder();
+        const ourInstallerDownloadFolder = IGlobalInstaller.getDownloadedInstallFilesFolder(installerUrl);
         this.file.wipeDirectory(ourInstallerDownloadFolder, this.acquisitionContext.eventStream);
         const installerPath = path.join(ourInstallerDownloadFolder, `${installerUrl.split('/').slice(-1)}`);
 
         const installerDir = path.dirname(installerPath);
         if (!fs.existsSync(installerDir)){
-            fs.mkdirSync(installerDir);
+            fs.mkdirSync(installerDir, {recursive: true});
         }
 
         await this.webWorker.downloadFile(installerUrl, installerPath);
         try
         {
             fs.chmodSync(installerPath, 0o744);
+
+            if(os.platform() === 'win32') // Windows does not have chmod +x ability with nodejs.
+            {
+                const commandRes = await this.commandRunner.execute(CommandExecutor.makeCommand('icacls', ['/grant', `"${installerPath}"`, `"%username%":F`, '/t', '/c']), null, false);
```

Comment:
Are these parameters ordered? According to https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/icacls#syntax the syntax should be `icacls <filename> [/grant[:r] <sid>:<perm>[...]]`

Also note that because you're not using `grant:r` you will be adding and retaining existing perms, which is different from the explicit 744 that chmod is doing. Also, should the chmod be moved into a separate `if` for non-windows and then explicitly set the ACL.

Can you share the results of this? Run the code, then from Powershell, run `Get-Acl <path-to-filename> | Select-Object -ExpandProperty Access` and just strip out any PII before pasting in it (or ping me offline).